### DISCUSSION TITLE
Interpretation of weight for pagerank(), betweenness_centrality(), closeness_centrality(), and clustering()

### DISCUSSION THREAD

**Discussion by mawic** [networkx/networkx#4733] [Q&A]
**Is a high weight of an edge equal to a short distance or vice versa?** 

I'm confused about how to interpret the weight/distance feature of the functions pagerank(), betweenness_centrality(), closeness_centrality(), and clustering().

- pagerank_numpy() uses weight
- betweenness_centrality() uses weight
- closeness_centrality() uses distance
- clustering() uses weight

Background: I model a social network and my edges have weights. The edge weight is the sum of interactions between two nodes. That means the higher the weight is, the shorter the distance. But I do not know whether I have to use the reciprocal of the weight as input for the functions or not.

**Comment by MridulS** [comment]
It depends if a higher weight/distance represents a "stronger" connection or a "weaker" connection. It also helps to get an intuition behind the algorithm if you want to use the weighted version.

For example if using weighted betweenness centrality to find the most "important" nodes you should take inverse weight, betweenness centrality calculates shortest paths between pairs of nodes so it would "avoid" the nodes which are "strongly" connected (i.e. have high weights/interactions). For PageRank it would be opposite, larger weights are better! 
There is some more explantation in this tutorial https://ericmjl.github.io/Network-Analysis-Made-Simple/05-casestudies/01-gameofthrones/#betweeness-centrality

**Reply by mawic** [reply]
Thanks, then I assume this also affects closeness_centrality.

**Reply by MridulS** [reply]
Yeah, check out the documentation at https://networkx.org/documentation/latest/reference/algorithms/generated/networkx.algorithms.centrality.closeness_centrality.html or better yet check out the code too https://github.com/networkx/networkx/blob/main/networkx/algorithms/centrality/closeness.py#L12 :)

**Comment by rossbar** [comment]
Just to add - the documentation was recently clarified for the centrality measures in #4610, so the interpretation of weight for the centrality measures should be clear in the development documentation: https://networkx.org/documentation/latest/. It seems a few other docstrings would benefit from a similar treatment - PRs with suggestions are welcome!

**Comment by BradKML** [comment]
✋ https://github.com/networkx/networkx/discussions/4808

Hello?

**Comment by BradKML** [comment]
Also about weighted networks it seemed that this software already has it https://www.semanticscholar.org/paper/CytoNCA%3A-A-cytoscape-plugin-for-centrality-analysis-Tang-Li/cd65fee1606720098d676b9e10c2014007454229
